uk.co.compucorp.civicrm.hrleaveandabsences/js/angular/src/leave-absences/absence-tab/components/absence-tab-work-patterns.js

```diff
@@ -10,22 +12,129 @@ define([
       return settings.pathTpl + 'components/absence-tab-work-patterns.html';
     }],
     controllerAs: 'workpatterns',
-    controller: ['$log', '$rootElement', '$uibModal', 'settings', controller]
+    controller: [
+      '$log', '$q', '$rootElement', '$uibModal', 'dialog',
+      'settings', 'OptionGroup', 'WorkPattern', controller]
   });
 
-  function controller($log, $rootElement, $uibModal, settings) {
+  function controller ($log, $q, $rootElement, $uibModal, dialog, settings, OptionGroup, WorkPattern) {
     $log.debug('Component: absence-tab-work-patterns');
 
-    var vm = {};
+    var changeReasons = [];
+    var vm = Object.create(this);
+
+    vm.customWorkpattern = [];
+    vm.defaultWorkPattern = null;
+    vm.loading = true;
```

Comment:
Please amend `vm.loading = true` to something like `vm.loading = { something: true }`